### PR TITLE
fix(insights): Support path cleaning rules with wildcard groups

### PR DESCRIPTION
**Pull request by Twixes**
## Problem

Combining path cleaning rules and wildcard groups made the query return empty results when a wildcard group was used for the start or end point. See this ticket for an example: ZEN-2794. The problem came from a conditional statement where we applied _either_ path cleaning rules or wildcard groups in the query.

## Changes

Now we always apply path cleaning rules (if there are any) and _then_ wildcard groups (if there are any).

## How did you test this code?

Added a query test.

### CODE REVIEW

File: posthog/queries/paths/paths_event_query.py [hunk 1]
```diff
@@ -150,29 +150,26 @@ def _get_grouping_fields(self) -> Tuple[List[str], Dict[str, Any]]:
...
                 params[f"alias_{idx}"] = alias
-
         else:
-            _fields.append("multiMatchAnyIndex(path_item_ungrouped, %(regex_groupings)s) AS group_index")
-            _fields.append("if(group_index > 0, %(groupings)s[group_index], path_item_ungrouped) AS path_item")
+            final_path_item_column = "path_item_ungrouped"
+
+        # Match wildcard groups
+        _fields.append(f"multiMatchAnyIndex({final_path_item_column}, %(regex_groupings)s) AS group_index")
+        _fields.append(f"if(group_index > 0, %(groupings)s[group_index], {final_path_item_column}) AS path_item")
```
Comment (Twixes) [inline posthog/queries/paths/paths_event_query.py:172]: This being pulled out is the key change

File: posthog/queries/paths/paths_event_query.py [hunk 2]
```diff
@@ -150,29 +150,26 @@ def _get_grouping_fields(self) -> Tuple[List[str], Dict[str, Any]]:
...
-                else:
-                    _fields.append(
-                        f"replaceRegexpAll(path_item_{idx - 1}, %(regex_replacement_{idx})s, %(alias_{idx})s) as path_item_{idx}"
-                    )
+                source_path_item_column = "path_item_ungrouped" if idx == 0 else f"path_item_{idx-1}"
+                result_path_item_column = "path_item_cleaned" if idx == len(replacements) - 1 else f"path_item_{idx}"
+                _fields.append(
+                    f"replaceRegexpAll({source_path_item_column}, %(regex_replacement_{idx})s, %(alias_{idx})s) "
+                    f"AS {result_path_item_column}"
+                )
```
Comment (Twixes) [inline posthog/queries/paths/paths_event_query.py:164]: This refactor is just to reduce the complexity of the code, works the same

**Review by neilkakkar — APPROVED** [review]
Seems alright! Makes sense

**Review by neilkakkar — APPROVED** [review]
Seems alright! Makes sense

**Pull request merged**